pkg/solana/relay.go

```diff
@@ -130,12 +137,32 @@ func (r *Relayer) NewConfigProvider(ctx context.Context, args relaytypes.RelayAr
 	return configWatcher, err
 }
 
-func (r *Relayer) NewContractWriter(_ context.Context, _ []byte) (relaytypes.ContractWriter, error) {
-	return nil, errors.New("contract writer is not supported for solana")
+func (r *Relayer) NewContractWriter(_ context.Context, config []byte) (relaytypes.ContractWriter, error) {
+	cfg := chainwriter.ChainWriterConfig{}
+	if err := json.Unmarshal(config, &cfg); err != nil {
+		return nil, fmt.Errorf("failed to unmarshall chain writer config err: %s", err)
+	}
+
+	solanaReader, err := r.chain.Reader()
```

Comment:
Realizing now we should be using the MultiClient here - but I can add that in later